### PR TITLE
Fix DateTime binding

### LINKED ISSUE #21687: Azure function service bus queue trigger parameter "EnqueuedTimeUtc" [BUG]
**Describe the bug**
Upgrading from extension version 4.3.0 to 5.0.0-beta.3 the DateTime EnqueuedTimeUtc param is in wrong string format.

**Expected behavior**
5.x to have the same string format as in 4.3.0.

**Actual behavior (include Exception or Stack Trace)**
In 4.3.0 string format is: EnqueuedTimeUtc: 2021-06-07T14:14:16.2230000Z
in 5.0.0-beta.3 string format is: EnqueuedTimeUtc: 2021-06-07T13:46:31.6680000+00:00

This raises exception "Exception binding parameter 'enqueuedTimeUtc'. Microsoft.Azure.WebJobs.Host: Binding data for 'enqueuedTimeUtc' is not of expected type DateTime."

This happens both in local environment and in Azure func app.

**To Reproduce**

1. Create new func app with service bus queue trigger.
`        
[FunctionName("SbQueueTest")]
public static async Task RunAsync([ServiceBusTrigger("test-queue", Connection = "ServiceBusConnection")]
  string myQueueItem, DateTime enqueuedTimeUtc, ILogger log)
  {
     log.LogInformation($"C# ServiceBus queue trigger function processed message: {myQueueItem}");
   }
`
2. Send message to queue
3. Receive message with 4.3.0 no problem
4. Receive message with 5.0.0-beta.3 with exception thrown

**Environment:**
 - Packages: microsoft.azure.webjobs.extensions.servicebus\4.3.0 or 5.0.0-beta.3
 - Windows 10, .NET SDK 5.0.101
 - IDE and version : JetBrains Rider 2021.1.3

### PR DESCRIPTION
**Pull request by JoshLove-msft**
Fixes https://github.com/Azure/azure-sdk-for-net/issues/21687

### CODE REVIEW

File: sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Triggers/ServiceBusTriggerBindingStrategy.cs [hunk 1]
```diff
@@ -130,8 +130,7 @@ internal static void AddBindingData(Dictionary<string, object> bindingData, Serv
                 deliveryCounts[i] = messages[i].DeliveryCount;
                 deadLetterSources[i] = messages[i].DeadLetterSource;
                 lockTokens[i] = messages[i].LockToken;
-                //this is temporary until the Service Bus SDK addresses the missing timezone issue in case DateTime.MaxValue, github.com/Azure/azure-sdk-for-net/issues/15343
-                expiresAtUtcs[i] = messages[i].ExpiresAt.DateTime.ToUniversalTime();
```
Comment (JoshLove-msft) [inline sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Triggers/ServiceBusTriggerBindingStrategy.cs:134]: This is not an issue in the new SDK as we use DateTimeOffset.MaxValue, which is in UTC.

**Review by jsquire — APPROVED** [review]

**Review by pakrym — APPROVED** [review]

**Pull request merged**